core/esmf-aspect-model-document-generators/src/main/java/org/eclipse/esmf/aspectmodel/generator/asyncapi/AspectModelAsyncApiGenerator.java

```diff
@@ -0,0 +1,299 @@
+import org.eclipse.esmf.metamodel.Property;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+
+public class AspectModelAsyncApiGenerator {
+
+   private static final String APPLICATION_JSON = "application/json";
+   private static final String CHANNELS = "#/channels";
+   private static final String COMPONENTS_SCHEMAS_PATH = "#/components/schemas";
+   private static final String COMPONENTS_MESSAGES = "#/components/messages";
+   private static final String ACTION_RECEIVE = "receive";
+   private static final String ACTION_SEND = "send";
+   private static final String V30 = "3.0.0";
+
+   private static final String TITLE_FIELD = "title";
+   private static final String DESCRIPTION_FIELD = "description";
+
+   private static final JsonNodeFactory FACTORY = JsonNodeFactory.instance;
+   private static final Logger LOG = LoggerFactory.getLogger( AspectModelAsyncApiGenerator.class );
+   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+   private enum JsonType {
+      NUMBER,
+      BOOLEAN,
+      STRING,
+      OBJECT;
+
+      private JsonNode toJsonNode() {
+         switch ( this ) {
+         case NUMBER:
+            return JsonNodeFactory.instance.textNode( "number" );
+         case BOOLEAN:
+            return JsonNodeFactory.instance.textNode( "boolean" );
+         case OBJECT:
+            return JsonNodeFactory.instance.textNode( "object" );
+         default:
+            return JsonNodeFactory.instance.textNode( "string" );
+         }
+      }
+   }
+
+   private static final Map<Resource, JsonType> TYPE_MAP = ImmutableMap.<Resource, JsonType> builder()
+         .put( XSD.xboolean, JsonType.BOOLEAN )
+         .put( XSD.decimal, JsonType.NUMBER )
+         .put( XSD.integer, JsonType.NUMBER )
+         .put( XSD.xfloat, JsonType.STRING )
+         .put( XSD.xdouble, JsonType.NUMBER )
+         .put( XSD.xbyte, JsonType.NUMBER )
+         .put( XSD.xshort, JsonType.NUMBER )
+         .put( XSD.xint, JsonType.NUMBER )
+         .put( XSD.xlong, JsonType.NUMBER )
+         .put( XSD.unsignedByte, JsonType.NUMBER )
+         .put( XSD.unsignedShort, JsonType.NUMBER )
+         .put( XSD.unsignedInt, JsonType.NUMBER )
+         .put( XSD.unsignedLong, JsonType.NUMBER )
+         .put( XSD.positiveInteger, JsonType.NUMBER )
+         .put( XSD.nonPositiveInteger, JsonType.NUMBER )
+         .put( XSD.negativeInteger, JsonType.NUMBER )
+         .put( XSD.nonNegativeInteger, JsonType.NUMBER )
+         .put( RDF.langString, JsonType.STRING )
+         .build();
+
+   public JsonNode applyForJson( final Aspect aspect, final String applicationId, final boolean useSemanticVersion,
```

Comment:
So, after some discussion with @ysrbo we decided to change some logic:

- Move general logic for generation separate file in separate GeneratorHelper.java;
- Put in generator just one method for getting `Map<Path, JsonNode>` structure for separate files `getContentWithSeparateSchemas`, it means, that each logic for generation some structure for YAML and separate files placed in server side;
- move logic for generation YAML files in `AspectToAsyncapiCommand`, because it looks like "client side" and our "client" can make everything with `Map<Path, JsonNode>` structure on his side.

Advantages:
- generation needed structures placed on the "server" side `AspectModelAsyncApiGenerator`;
- needed manipulation with provided structure `Map<Path, JsonNode>` placed on the "client" side `AspectToAsyncapiCommand`;
- `AsyncApiSchemaArtifact` plays role DataStructure.

Thanks,
Best Regards!